src/api_manager/request_handler.cc

```diff
@@ -49,8 +50,27 @@ RequestHandler::RequestHandler(
 
 void RequestHandler::Check(std::function<void(Status status)> continuation) {
   auto interception = [continuation, this](Status status) {
-    if (status.ok() && context_->cloud_trace()) {
-      context_->StartBackendSpanAndSetTraceContext();
+    if (status.ok()) {
+      if (context_->cloud_trace()) {
+        context_->StartBackendSpanAndSetTraceContext();
+      }
+      // Add InstanceIdentityToken to request header if needed, which is
+      // required for backend routing.
+      std::string audience;
+      if (context_->method()) {
+        audience = context_->method()->backend_jwt_audience();
```

Comment:
not to do string copy

Reply:
maybe to put all these token logic into request_context

Reply:
Done.